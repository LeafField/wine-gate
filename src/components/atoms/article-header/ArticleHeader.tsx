import React, { FC } from "react";

type Props = {
  title: string;
};

const ArticleHeader: FC<Props> = ({ title }) => {
  return (
    <h2 className="article-title mx-4 mt-8 border-l-4 border-blue py-[0.125rem] pl-2 text-heading2 tablet:mx-0 tablet:pt-0">
      {title}
    </h2>
  );
};

export default ArticleHeader;
