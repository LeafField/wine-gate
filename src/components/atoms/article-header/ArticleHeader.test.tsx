import { screen, render } from "@testing-library/react";
import ArticleHeader from "./ArticleHeader";

describe("ArticleHeaderの単体テスト", () => {
  test("propsとして受け取ったtitleが表示される", () => {
    render(<ArticleHeader title="テストタイトル" />);
    expect(screen.getByText("テストタイトル")).toBeInTheDocument();
  });

  test("スナップショットテスト", () => {
    const { container } = render(
      <ArticleHeader title="コノスル カベルネソーヴィニヨン ビシクレタ・レゼルバ" />,
    );
    expect(container).toMatchSnapshot();
  });
});
