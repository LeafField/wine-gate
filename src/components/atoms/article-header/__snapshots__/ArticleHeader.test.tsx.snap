// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleHeaderの単体テスト スナップショットテスト 1`] = `
<div>
  <h2
    class="article-title mx-4 mt-8 border-l-4 border-blue py-[0.125rem] pl-2 text-heading2 tablet:mx-0 tablet:pt-0"
  >
    コノスル カベルネソーヴィニヨン ビシクレタ・レゼルバ
  </h2>
</div>
`;
